Nos/Views/CreateUsernameSheet.swift

```diff
@@ -0,0 +1,407 @@
+        NavigationStack {
+            ClaimYourUniqueIdentityPage(isPresented: $isPresented)
+        }
+        .frame(idealWidth: 320, idealHeight: 480)
+        .presentationDetents([.medium])
+    }
+}
+
+fileprivate class UsernameObserver: ObservableObject {
+
+    @Published
+    var debouncedText = ""
+
+    @Published
+    var text = ""
+
+    private var subscriptions = Set<AnyCancellable>()
+
+    init() {
+        $text
+            .removeDuplicates()
+            .filter { $0.count >= 3 }
+            .debounce(for: .milliseconds(200), scheduler: RunLoop.main)
+            .sink { [weak self] value in
+                self?.debouncedText = value
+            }
+            .store(in: &subscriptions)
+    }
+}
+
+fileprivate struct ClaimYourUniqueIdentityPage: View {
+
+    @Binding var isPresented: Bool
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 20) {
+            Spacer(minLength: 40)
+            PlainText(String(localized: LocalizedStringResource.localizable.new).uppercased())
+                .padding(.horizontal, 6)
+                .padding(.vertical, 5)
+                .font(.clarity(.bold, textStyle: .footnote))
+                .foregroundStyle(Color.white)
+                .background {
+                    Color.secondaryTxt
+                        .cornerRadius(4, corners: .allCorners)
+                }
+            PlainText(.localizable.claimUniqueUsernameTitle).sheetTitle()
+            PlainText(.localizable.claimUniqueUsernameDescription).sheetDescription()
+
+            Spacer(minLength: 0)
+
+            NavigationLink(String(localized: LocalizedStringResource.localizable.setUpMyUsername)) {
+                PickYourUsernamePage(isPresented: $isPresented)
+            }
+            .buttonStyle(BigActionButtonStyle())
+
+            Spacer(minLength: 40)
+        }
+        .padding(.horizontal, 40)
+        .sheetPage()
+    }
+}
+
+fileprivate struct PickYourUsernamePage: View {
+
+    @Binding var isPresented: Bool
+    @StateObject private var usernameObserver = UsernameObserver()
+    @State private var verified: Bool?
+    @State private var isVerifying = false
+    @FocusState private var usernameFieldIsFocused: Bool
+    @Dependency(\.namesAPI) var namesAPI
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 20) {
+            Button {
+                isPresented = false
+            } label: {
+                PlainText(.localizable.cancel)
+                    .font(.clarity(.medium, textStyle: .body))
+                    .foregroundStyle(Color.primaryTxt)
+                    .padding()
+            }
+            ScrollView(.vertical, showsIndicators: false) {
+                VStack(alignment: .leading, spacing: 20) {
+                    PlainText(.localizable.pickYourUsernameTitle).sheetTitle()
+                    PlainText(.localizable.pickYourUsernameDescription).sheetDescription()
+                    HStack {
```

Comment:
I might like to see this HStack or the above VStack refactored out into a var, func, or struct so it's clearer to me what it contains. I guess this is mainly one TextField but with all the modifiers it's a little hard for me to follow.